src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/AddAatfSiteRequestHandler.cs

```diff
@@ -15,28 +17,33 @@ internal class AddAatfSiteRequestHandler : IRequestHandler<AddAatfSite, bool>
         private readonly IWeeeAuthorization authorization;
         private readonly IAddAatfSiteDataAccess offSiteDataAccess;
         private readonly IGenericDataAccess genericDataAccess;
+        private readonly IOrganisationDetailsDataAccess organisationDetailsDataAccess;
 
         public AddAatfSiteRequestHandler(WeeeContext context, IWeeeAuthorization authorization,
-            IAddAatfSiteDataAccess offSiteDataAccess, IGenericDataAccess genericDataAccess)
+            IAddAatfSiteDataAccess offSiteDataAccess, IGenericDataAccess genericDataAccess,
+            IOrganisationDetailsDataAccess organisationDetailsDataAccess)
         {
             this.context = context;
             this.authorization = authorization;
             this.offSiteDataAccess = offSiteDataAccess;
             this.genericDataAccess = genericDataAccess;
+            this.organisationDetailsDataAccess = organisationDetailsDataAccess;
         }
 
         public async Task<bool> HandleAsync(AddAatfSite message)
         {
             authorization.EnsureCanAccessExternalArea();
 
+            Country country = await organisationDetailsDataAccess.FetchCountryAsync(message.AddressData.CountryId);
```

Comment:
Add an additional test to check that the country is retreived and the AatfAddress object country is property is set with the retrieved value